### PR TITLE
Add require_contact_method and contact_method args for ServiceOwnersh…

### PR DESCRIPTION
**Pull request by maxim-opslevel**
opslevel-go needs to be at least commit version `v2022.7.21-0.20220722020835-ea98e9aa020c` to work.

Test scenarios

- Create
  - ServiceOwnershipCheck with no require_contact_method or contact_method args
  - ServiceOwnershipCheck with just require_contact_method arg
  - ServiceOwnershipCheck with just contact_method arg
  - ServiceOwnershipCheck with require_contact_method=true and contact_method arg set to one of the values

- Update
  - ServiceOwnershipCheck with no require_contact_method or contact_method args changes nothing
  - ServiceOwnershipCheck with just require_contact_method arg just changes require_contact_method
  - ServiceOwnershipCheck with just contact_method arg just changes contact_method
  - Removing previously set require_contact_method sets it to false
  - Removing previously set contact_method sets it to "ANY"

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -62,6 +62,21 @@ There are some tools that will be helpful to you in developing locally. While th
...
+## Using a local version of opslevel-go
+
+In `go.mod` uncomment (remove the //) the line at the bottom of the file:
+
+```
+// Uncomment for local development
+//replace github.com/opslevel/opslevel-go => ./submodules/opslevel-go/
+```
+
+Now you may make changes in your local git submodule of `opslevel-go`.
```
Comment (maxim-opslevel) [inline CONTRIBUTING.md:74]: This worked for me at one point but stopped working. Not sure why? Am I missing something from the instructions?

**Comment by maxim-opslevel** [comment]
Question. For the scenario where a previously set param is removed, and thus we are setting it to null in Terraform but sending the GraphQL API some default value, should the Terrraform preview reflect that. 

e.g.

```
Terraform will perform the following actions:

  # opslevel_check_service_ownership.example will be updated in-place
  ~ resource "opslevel_check_service_ownership" "example" {
      - contact_method         = "SLACK" -> null
        id                     = "Z2lkOi8vb3BzbGV2ZWwvQ2hlY2tzOjpIYXNPd25lci8xNTc"
        name                   = "TERRAFORM CREATED V3"
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

should this instead be:

```
Terraform will perform the following actions:

  # opslevel_check_service_ownership.example will be updated in-place
  ~ resource "opslevel_check_service_ownership" "example" {
      - contact_method         = "SLACK" -> "ANY
        id                     = "Z2lkOi8vb3BzbGV2ZWwvQ2hlY2tzOjpIYXNPd25lci8xNTc"
        name                   = "TERRAFORM CREATED V3"
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (rocktavious) [reply CONTRIBUTING.md:74]: @maxim-opslevel oh ya it now needs to be
```
replace github.com/opslevel/opslevel-go/v2022 => ./submodules/opslevel-go/
```
the package name changed to appease the golang version gods

**Comment by rocktavious** [comment]
Ya so this is the tricky part - So its my understanding that if we submit an update call to the API where `contact_method` == null then the default is `any` ?

If thats the case then yes the first terraform diff is good - but we need to be careful not the put the value `any` into the state if the user hasn't specified `contact_method` so that you don't run into an infinite diff always trying to make 
```
- contact_method         = "ANY" -> null
```

Does that make sense.  This is a very ticky tacky thing with Terraform.

File: opslevel/resource_opslevel_check_service_ownership.go [hunk 2]
```diff
@@ -44,13 +67,40 @@ func resourceCheckServiceOwnershipRead(d *schema.ResourceData, client *opslevel.
...
 	setCheckUpdateInput(d, &input)
 
+	if d.HasChange("require_contact_method") {
+		input.RequireContactMethod = opslevel.Bool(d.Get("require_contact_method").(bool))
+	}
+
+	if d.HasChange("contact_method") {
+		contactMethod := opslevel.ServiceOwnershipCheckContactType(d.Get("contact_method").(string))
+
+		if contactMethod == "" {
```
Comment (rocktavious) [inline opslevel/resource_opslevel_check_service_ownership.go:96]: This part could be simplifed to

```
if contactMethod == "" {
	contactMethod = opslevel.ServiceOwnershipCheckContactType("any")
} 
input.ContactMethod = &contactMethod
```

**Review by rocktavious — CHANGES_REQUESTED** [review]
Just 1 documentation change and 1 logic simplification and then I think its good.  Thanks for all the hard work @maxim-opslevel

**Comment by rocktavious** [comment]
To get tests to pass we'll need to change go.mod to use the submodule

**Comment by maxim-opslevel** [comment]
> To get tests to pass we'll need to change go.mod to use the submodule

Are we okay changing go.mod to use a specific commit instead of a Release?

**Comment by rocktavious** [comment]
yep - thats the point of it - so we can do WIP to both tools

**Comment by maxim-opslevel** [comment]
Cool, I've addressed the suggestions in my latest commit. I've also added another small section to CONTRIBUTION.md

**Review by rocktavious — APPROVED** [review]
LGTM

**Pull request merged**